lua/entities/gmod_wire_expression2/core/wirelink.lua

```diff
@@ -107,6 +107,17 @@ local function WriteArray(entity, address, data)
 	return free_address
 end
 
+local function writeArraySimple(entity, address, data)
+	local written = 0
+	for index, value in pairs(data) do
+		if type(value) == "number" then
```

Comment:
That is correct, it doesn't need to be checked. and the output converter is not relevant here because this code doesn't deal with vectors or angles, just numbers